feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/common/RealReferendaConstructor.kt

```diff
@@ -289,6 +286,52 @@ class RealReferendaConstructor(
         }
     }
 
+    private fun confirmingStatusFromOnChain(
+        status: OnChainReferendumStatus.Ongoing,
+        blockDurationEstimator: BlockDurationEstimator,
+        track: TrackInfo,
+        referendumId: ReferendumId,
+        votingByReferenda: Map<ReferendumId, ReferendumVoting?>
+    ): ReferendumStatus.Ongoing {
+        val decidingStatus = status.deciding!!
+        val referendumVoting = votingByReferenda[referendumId]
+        val delayedPassing = referendumVoting?.delayedPassing()
+
+        val isCurrentlyPassing = referendumVoting?.currentlyPassing().orFalse()
+        val isPassingAfterDelay = delayedPassing != null && delayedPassing.passingInFuture
+
+        return when {
+            isCurrentlyPassing || isPassingAfterDelay -> {
+                val confirmingStatus = decidingStatus.confirming.cast<ConfirmingSource.OnChain>().status
+                when {
+                    // Confirmation period started
+                    confirmingStatus != null -> {
+                        val approveBlock = confirmingStatus.till
+                        val approveIn = blockDurationEstimator.timerUntil(approveBlock)
+
+                        ReferendumStatus.Ongoing.Confirming(approveIn = approveIn)
+                    }
+
+                    // Deciding period that will be approved in delay
+                    else -> {
+                        val delay = delayedPassing!!.delayFraction
+                        val approveBlock = decidingStatus.since.toBigDecimal() + delay * track.decisionPeriod.toBigDecimal()
+                        val approveIn = blockDurationEstimator.timerUntil(approveBlock.toBigInteger())
+
+                        ReferendumStatus.Ongoing.DecidingApprove(confirmingIn = approveIn)
+                    }
+                }
+            }
```

Comment:
Seems like this could be simplified:

``` kotlin
when {
isCurrentlyPassing -> {
 // Confirmation period started block
}
isPassingAfterDelay -> {
///Deciding period that will be approved in delay block
}
else -> {
// reject block
}
}
```